wp-content/plugins/core/src/Templates/Content/Panels/Wysiwyg.php

```diff
@@ -14,15 +20,64 @@ public function get_data(): array {
 		return $data;
 	}
 
+	/**
+	 * Get the mapped panel data.
+	 *
+	 * @return array
+	 */
 	public function get_mapped_panel_data(): array {
 		$data = [
-			'title'   => $this->get_title( $this->panel_vars[ Wysi::FIELD_TITLE ], [ 's-title', 'h2' ] ),
-			'columns' => $this->get_the_columns(),
+			'title'           => $this->get_title( $this->panel_vars[ Wysi::FIELD_TITLE ], [ 's-title', 'h2' ] ),
+			'wrapper_classes' => $this->get_wrapper_classes(),
+			'columns'         => $this->get_the_columns(),
 		];
 
 		return $data;
 	}
 
+	/**
+	 * Overrides `get_classes()` from the Panel parent class.
+	 *
+	 * Return value is available in the twig template via the `classes` twig variable in the parent class.
+	 *
+	 * @return string
+	 */
+	protected function get_classes(): string {
+		$classes = [
+			'panel',
+			's-wrapper',
+			'site-panel',
+			sprintf( 'site-panel--%s', $this->panel->get_type_object()->get_id() ),
+			sprintf( 'site-panel--wysiwyg__layout-%s', $this->panel_vars[ Wysi::FIELD_LAYOUT ] ),
+		];
+		return Util::class_attribute( $classes );
+	}
+
+	/**
+	 * Get WYSIWYG row wrapper classes.
+	 *
+	 * @return string
+	 */
+	protected function get_wrapper_classes() {
+		$classes = [];
+		$columns = $this->panel_vars[ Wysi::FIELD_COLUMNS ];
+
```

Comment:
Let's check if $columns is empty here and return `Util::class_attribute( $classes, false )` if so. Then we can continue without having to check that in the next to conditionals.

Reply:
@speerface is that needed? Saw this on the project where it's getting implemented and seems like it's redundant given you still end up with the same return statement.

https://github.com/moderntribe/wstc/pull/1#discussion_r313949859